src/Compilers/VisualBasic/Portable/Semantics/Conversions.vb

```diff
@@ -1261,12 +1263,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     Return Nothing 'ConversionKind.NoConversion
                 End If
 
+                involvesNarrowingFromNumericConstant = involvesNarrowingFromNumericConstant Or elementConversion
```

Comment:
I would either rename the "involvesNarrowing..." and "allNarrowing..." locals to "involvedConversions" and "commonConversions", or apply the mask `ConversionKind.InvolvesNarrowingFromNumericConstant` up here.